DNN Platform/Library/Services/ClientDependency/Extensions.cs

```diff
@@ -0,0 +1,172 @@
+            }
+
+            if (checkIfExists)
+            {
+                var physicalPath = HttpContext.Current.Server.MapPath(scriptPath);
+                if (!System.IO.File.Exists(physicalPath))
+                {
+                    return;
+                }
+            }
+
+            controller.CreateScript()
+                .FromSrc(scriptPath)
+                .SetPriority(priority)
+                .Register();
+        }
+
+        /// <summary>
+        /// Registers a stylesheet resource by its path.
+        /// </summary>
+        /// <param name="controller">The <see cref="IClientResourceController"/> to extend.</param>
+        /// <param name="stylesheetPath">The path to the stylesheet resource to register.</param>
+        /// <param name="checkIfExists">Whether to check if the script already exists before registering.</param>
+        public static void RegisterStylesheet(this IClientResourceController controller, string stylesheetPath, bool checkIfExists = false)
+        {
+            if (controller == null)
+            {
+                throw new System.ArgumentNullException(nameof(controller));
+            }
+
+            if (string.IsNullOrEmpty(stylesheetPath))
+            {
+                throw new System.ArgumentException("Value cannot be null or empty.", nameof(stylesheetPath));
+            }
+
+            if (checkIfExists)
+            {
+                var physicalPath = HttpContext.Current.Server.MapPath(stylesheetPath);
+                if (!System.IO.File.Exists(physicalPath))
+                {
+                    return;
+                }
+            }
+
+            controller.CreateStylesheet().FromSrc(stylesheetPath).Register();
+        }
+
+        /// <summary>
+        /// Registers a stylesheet resource by its path.
+        /// </summary>
+        /// <param name="controller">The <see cref="IClientResourceController"/> to extend.</param>
+        /// <param name="stylesheetPath">The path to the stylesheet resource to register.</param>
+        /// <param name="priority">The priority for loading the stylesheet.</param>
+        /// <param name="checkIfExists">Whether to check if the script already exists before registering.</param>
+        public static void RegisterStylesheet(this IClientResourceController controller, string stylesheetPath, FileOrder.Css priority, bool checkIfExists = false)
+        {
+            if (controller == null)
+            {
+                throw new System.ArgumentNullException(nameof(controller));
+            }
+
+            if (string.IsNullOrEmpty(stylesheetPath))
+            {
+                throw new System.ArgumentException("Value cannot be null or empty.", nameof(stylesheetPath));
+            }
+
+            if (checkIfExists)
+            {
+                var physicalPath = HttpContext.Current.Server.MapPath(stylesheetPath);
```

Comment:
prefer [`HostingEnvironment.MapPath`](https://learn.microsoft.com/en-us/dotnet/api/system.web.hosting.hostingenvironment.mappath) over `Server.MapPath`, since it doesn't depend on `HttpContext`